### ISSUE TITLE
Tensorflow Version Syntax Mismatched

### ISSUE THREAD

**Issue by mlearning** [haeusser/learning_by_association#7]
In backend.py line 70, the tf.concat has the new signature:   

```python
  tf.concat(batch_images, 0), tf.concat(batch_labels, 0)
```

In tarin.py line 301 and 309, it has the old signature:
```python
  t_sup_emb = tf.concat(0, [
                    t_sup_emb, semisup.create_virt_emb(FLAGS.virtual_embeddings,
                                                       FLAGS.emb_size)
                ])
```

Is this meant for Tensorflow 1.0 or above?  Could you please kindly list the dependencies in the readme file, such as Python version, Tensorflow Version and Numpy version etc? 

When I ran it with Python 3.5 and Tensorflow 1.1, I ran into a problem when reading the stl-10 bin file:

UnicodeDecodeError: 'utf-8' codec can't decode byte 0x92 in position 0: invalid start byte  
File "/root/DL/computer-vision/learning_by_association/semisup/tools/stl10.py", line 67, in extract_images
```python
 imgs = np.fromstring(f.read(), np.uint8)  
```

So I ran it with Python 2.7 and tensorflow 0.12.1 by changing the above tf.concat signature to the old version, and encounter another error: 
backend.py, line 179, in add_semisup_loss         
```python
    loss_aba = tf.losses.softmax_cross_entropy(    
```
AttributeError: 'module' object has no attribute 'losses'       

This is an indication that the backend.py is using Tensorflow version 1.0 or higher.

However in some other  places I see indications of lower version being used.  Your clarification or fixes will be very appreciated!

**Comment by mlearning** [comment]
I figured out that it is with Python 2.7 and Tensorflow 1.0 or above.  I am able to start the training process now. 

The tf.concat in training.py with the wrong signature is not throwing an error because virtual embedding is not enabled.

**Comment by haeusser** [comment]
I think the issue you're running into is not related to TF but to reading the STL10 data. Did you download the data set, unzipped it and adjusted the path in `semisup/tools/data_dirs.py`?

I should fix the tf.concat calls.

**Comment by haeusser** [comment]
fixed in bfe53989f9883610f7a6ec2322b2c6177dc1524b
